Library/Homebrew/cask/download.rb

```diff
@@ -9,71 +10,75 @@ module Cask
   # A download corresponding to a {Cask}.
   #
   # @api private
-  class Download
+  class Download < ::Downloadable
     include Context
 
     attr_reader :cask
 
     def initialize(cask, quarantine: nil)
+      super()
+
       @cask = cask
       @quarantine = quarantine
     end
 
+    sig { override.returns(T.nilable(::URL)) }
+    def url
+      @url ||= ::URL.new(cask.url.to_s, cask.url.specs)
+    end
+
+    sig { override.returns(T.nilable(::Checksum)) }
+    def checksum
+      @checksum ||= cask.sha256 if cask.sha256 != :no_check
+    end
+
+    sig { override.returns(T.nilable(::Version)) }
+    def version
+      @version ||= ::Version.create(cask.version)
+    end
+
+    sig {
+      override
```

Comment:
What's this do/for?

Reply:
If the upstream type signature changes then this will error so that we make sure changes are reflected in subclasses too.

Reply:
Nice, gotcha!